compiler/ballerina-treegen/src/main/java/io/ballerinalang/compiler/internal/treegen/targets/Target.java

```diff
@@ -55,16 +57,18 @@ public Target(TreeGenConfig config) {
         this.config = config;
     }
 
-    public abstract List<SourceText> execute(SyntaxTree syntaxTree);
+    public abstract List<SourceText> execute(SyntaxTree syntaxTree, HashMap<String, SyntaxNodeMetadata> templateConfig);
 
     protected abstract String getTemplateName();
 
     protected TreeNodeClass convertToTreeNodeClass(SyntaxNode syntaxNode,
                                                    String packageName,
-                                                   List<String> importClassNameList) {
+                                                   List<String> importClassNameList,
+                                                   HashMap<String, SyntaxNodeMetadata> templateConfig) {
```

Comment:
May be
```suggestion
                                                   HashMap<String, SyntaxNodeMetadata> nodeMetadataMap) {
```

Let's fix the variable names used in the other places too according to the new name.